### PR TITLE
Alertmanager ttl

### PR DESCRIPTION
**Pull request by aerosouund**
## Description

Add TTL field to alertmanager config
solves: https://github.com/prometheus-operator/prometheus-operator/issues/6360

## Type of change

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [x] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Verification

Please check the [Prometheus-Operator testing guidelines](../TESTING.md) for recommendations about automated tests.

## Changelog entry

```release-note
add ttl to alertmanagerConfig
```

### CODE REVIEW

**Comment by yp969803** [comment]
@aerosouund  you have added the field that is good,
Some reviews:
-  Run "go fmt -w ." after the completion of code
- Run "make generate"
- Merge all the commit into one commit which should follow a certain pattern you can see previous commits

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/prometheus-operator/prometheus-operator
 
-go 1.22
+go 1.22.0
```
Comment (ArthurSens) [inline go.mod:3]: Was this change intentional?

File: pkg/alertmanager/types.go [hunk 2]
```diff
@@ -29,6 +29,7 @@ import (
 // https://github.com/prometheus/alertmanager/issues/1985
 type alertmanagerConfig struct {
 	Global            *globalConfig   `yaml:"global,omitempty" json:"global,omitempty"`
+	TTL 			  duration        `yaml:"ttl,omitempty" json:"ttl,omitempty"`
```
Comment (ArthurSens) [inline pkg/alertmanager/types.go:32]: We have a linting problem here, you can fix it by running `make format` :)

File: pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go [hunk 3]
```diff
@@ -783,6 +784,9 @@ type PushoverConfig struct {
 	// Notification title.
 	// +optional
 	Title string `json:"title,omitempty"`
+	// time to live for the alert notification
+	// +optional
+	TTL time.Duration `json:"ttl,omitempty"`
```
Comment (ArthurSens) [inline pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go:789]:
```suggestion
	TTL *Duration `json:"ttl,omitempty"`
```

We have a [custom Duration type](https://github.com/prometheus-operator/prometheus-operator/blob/6e94864406d265774c1572b187e939f3ea5b8a3e/pkg/apis/monitoring/v1/types.go#L41), which has some useful validations on the API level. Could we use this one instead?

Also, we follow k8s API conventions as much as possible, which means we want to use [pointers for optional fields](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required).

**Review by ArthurSens — COMMENTED** [review]
Thanks for starting the PR!

The changes made to the API are almost there, we also want to incorporate the changes you've made to the API to the alertmanager packages, i.e. we want to write this configuration in Alertmanager's config files if it is set.

@slashpai shared some directions here: https://github.com/prometheus-operator/prometheus-operator/issues/6360#issuecomment-1983838148
